collectors/o365/o365_collector.js

```diff
@@ -168,7 +169,20 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
-            if (err.message) {
+            // set errorCode to showcase client error on DDMetric
```

Comment:
could you please extract the new code below into a separate function, like `_handleMaxPayloadExceededError`